### PR TITLE
Added script to generate local_test_coverage_report

### PR DESCRIPTION
**Pull request by hamza-yusuff**
## Notion ticket link

[Setup local code coverage reporting tools](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=861abcaffb604cd08d916638a41fd612)

## Implementation description
* Added a python script to run python coverage module over all the tests

## Steps to test
1. Run the command ``` docker exec -it cas_py_backend /bin/bash -c "python /app/tests/coverage.py"```

## What should reviewers focus on?
* A report like this should be generated --> 
( Note : the actual report is much bigger )

Screenshot 2022-07-03 at 11 55 50 AM

Screenshot 2022-07-03 at 11 57 26 AM
Screenshot 2022-07-03 at 11 57 41 AM

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by jennchenn — CHANGES_REQUESTED** [review]
exciting stuff!!

1. can you make this a shell script instead of importing `os`?
2. can you add `*.coverage` files to `.gitignore`
3. can you add some instructions for how to run the coverage script in the readme (add a subsection under the running tests section)

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@
 .vscode
 **/*.cache
 **/*.egg-info
+*.coverage
```
Comment (jennchenn) [inline .gitignore:12]: I think you'll also have to do a `git rm --cached *.coverage` in your terminal to ensure the committed `.coverage` files are ignored in the future

**Review by jennchenn — COMMENTED** [review]
looking good! few comments

I also had to add `coverage` to my `requirements.txt` file - can you double check whether or not you needed to do this and make changes as appropriate?

```
coverage==6.4.1
```

File: .gitignore (hunk 1 above)
Comment (hamza-yusuff) [reply .gitignore:12]: Thanks, I did this.

Comment (jennchenn) [reply .gitignore:12]: i think you'll have to commit the deleted files! i still see it in commit

File: README.md [hunk 2]
```diff
@@ -82,6 +82,13 @@ $ cd e2e-tests
 $ python3 -m pytest --lang python --auth --fs
 ```
 
+## Generating Test Coverage Report
+
+To generate a test coverage report, run:
+```
+ docker exec -it cas_py_backend /bin/bash -c "bash /app/tests/coverage.sh"
```
Comment (jennchenn) [inline README.md:89]:
```suggestion
 docker exec -it cas_py_backend /bin/bash -c "./tests/coverage.sh"
```

File: backend/python/tests/coverage.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```
Comment (jennchenn) [inline backend/python/tests/coverage.sh:1]: to stay consistent, let's use bash instead
```suggestion
#!/bin/bash
```

Comment (jennchenn) [reply backend/python/tests/coverage.sh:1]: you may have to toggle the `CRLF/LF` if you get a bad interpreter message:
image

**Review by jennchenn — COMMENTED** [review]
the command didn't output the report for me as is for some reason (i tested with a newly built container) - commented the changes i made to make it work!

File: .gitignore (hunk 1 above)
Comment (hamza-yusuff) [reply .gitignore:12]: Deleted the files from the tree. It should be okay i guess

**Review by jennchenn — COMMENTED** [review]
figured out how to make line endings consistent - can you add this to `.gitattributes`:
```
backend/python/tests/coverage.sh eol=lf
```

and run `git add --renormalize .` and commit?

after this both commands work for me (with / without bash)

File: .gitattributes [hunk 4]
```diff
@@ -1 +1,2 @@
 db-init/create-multiple-dbs.sh eol=lf
+backend/python/tests/coverage.sh eol=lf
```
Comment (jennchenn) [inline .gitattributes:2]: nit: newline

**Review by jennchenn — APPROVED** [review]
nit: i would prefer `docker exec -it cas_py_backend /bin/bash -c "./tests/coverage.sh"` if that command works for you as well

will approve now, so just merge when you fix the nit! nice work

**Pull request merged**